tests/bot/exts/moderation/test_incidents.py

```diff
@@ -768,3 +788,64 @@ async def test_on_message_non_incident(self):
             await self.cog_instance.on_message(MockMessage())
 
         mock_add_signals.assert_not_called()
+
+
+class TestMessageLinkEmbeds(TestIncidents):
+    """Tests for `extract_message_links` coroutine."""
+
+    async def extract_and_form_message_link_embeds(self):
+        """
+        Extract message links from a mocked message and form the message link embed.
+
+        Considers all types of message links, discord supports.
+        """
+        self.guild_id_patcher = mock.patch("bot.exts.backend.sync._cog.constants.Guild.id", 5)
+        self.guild_id = self.guild_id_patcher.start()
+
+        msg = MockMessage(id=555, content="Hello, World!" * 3000)
+        msg.channel.mention = "#lemonade-stand"
+
+        msg_links = [
+            # Valid Message links
+            f"https://discord.com/channels/{self.guild_id}/{msg.channel.discord_id}/{msg.discord_id}",
+            f"http://canary.discord.com/channels/{self.guild_id}/{msg.channel.discord_id}/{msg.discord_id}",
+
+            # Invalid Message links
+            f"https://discord.com/channels/{msg.channel.discord_id}/{msg.discord_id}",
+            f"https://discord.com/channels/{self.guild_id}/{msg.channel.discord_id}000/{msg.discord_id}",
+        ]
+
+        incident_msg = MockMessage(
+            id=777,
+            content=f"I would like to report the following messages, "
+                    f"as they break our rules: \n{', '.join(msg_links)}"
+        )
+
+        embeds = await extract_message_links(incident_msg, self.cog_instance.bot)
+        description = (
+            f"**Author:** {format_user(msg.author)}\n"
+            f"**Channel:** {msg.channel.mention} ({msg.channel.category}/#{msg.channel.name})\n"
+            f"**Content:** {('Hello, World!' * 3000)[:300] + '...'}\n"
```

Comment:
I still think you should slice before counting lines, but regardless you should probably add cases for the 50 chars and the 3 lines scenarios.

Reply:
Resolved in commit 488be0c.